### PR TITLE
PODS-1386 : PSA Reg - NON UK - Partnership - NAVIGATION

### PR DESCRIPTION
**Pull request by abhijeetardale**
PODS-1386 : PSA Reg - NON UK - Partnership - NAVIGATION

### CODE REVIEW

File: test/utils/NavigatorBehaviour.scala [hunk 1]
```diff
@@ -31,6 +38,12 @@ import scala.concurrent.ExecutionContext.Implicits.global
 trait NavigatorBehaviour extends PropertyChecks with OptionValues {
   this: WordSpec with MustMatchers =>
 
+  protected val emptyAnswers: UserAnswers = UserAnswers(Json.obj())
```
Comment (PaulCDurham) [inline test/utils/NavigatorBehaviour.scala]: This doesn't belong here. NavigatorBehaviour is a generic navigator test. Anything no-UK specific should be somewhere else, ie a non-UK specific test

File: app/utils/navigators/PartnershipNavigator.scala [hunk 2]
```diff
@@ -116,4 +124,29 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
       case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
-}
\ No newline at end of file
+
+  private def regionBasedNameNavigation(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(AreYouInUKId) match {
+      case Some(false) => NavigateTo.dontSave(routes.PartnershipRegisteredAddressController.onPageLoad())
+      case _ => NavigateTo.dontSave(routes.ConfirmPartnershipDetailsController.onPageLoad())
```
Comment (PaulCDurham) [inline app/utils/navigators/PartnershipNavigator.scala]: case _ could be Some(true) or None - should the behaviour be the same for both cases?

File: app/utils/navigators/PartnershipNavigator.scala [hunk 3]
```diff
@@ -116,4 +124,29 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
...
+    }
+  }
+
+  private def regionBasedNavigation(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(PartnershipRegisteredAddressId) flatMap { address =>
+      countryOptions.regions(address.country.getOrElse("")) match {
```
Comment (PaulCDurham) [inline app/utils/navigators/PartnershipNavigator.scala]: Sticking with an Option would be neater and more readable than getOrElse(""). Match on Some(UK) etc

File: app/utils/navigators/PartnershipNavigator.scala [hunk 4]
```diff
@@ -116,4 +124,29 @@ class PartnershipNavigator @Inject()(val dataCacheConnector: UserAnswersCacheCon
...
+        case RestOfTheWorld => NavigateTo.dontSave(routes.OutsideEuEeaController.onPageLoad())
+        case _ => NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
+      }
+    }
+  }
+
+  private def regionBasedContactDetailsRoutes(answers: UserAnswers): Option[NavigateTo] = {
+    answers.get(AreYouInUKId) match {
+      case Some(false) => NavigateTo.save(routes.CheckYourAnswersController.onPageLoad())
+      case _ => NavigateTo.save(routes.PartnershipVatController.onPageLoad(NormalMode))
```
Comment (PaulCDurham) [inline app/utils/navigators/PartnershipNavigator.scala]: case _ could be Some(true) or None - should the behaviour be the same for both cases?

File: test/utils/NavigatorBehaviour.scala (hunk 1 above)
Comment (abhijeetardale) [reply test/utils/NavigatorBehaviour.scala]: fixed

File: app/utils/navigators/PartnershipNavigator.scala (hunk 2 above)
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: fixed

File: app/utils/navigators/PartnershipNavigator.scala (hunk 4 above)
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: fixed

File: app/utils/navigators/PartnershipNavigator.scala (hunk 3 above)
Comment (abhijeetardale) [reply app/utils/navigators/PartnershipNavigator.scala]: Its not easy as we are using case object with optional field and  calling method needs string

**Pull request closed without merging**